admin/python/lsst/qserv/admin/watcherLib.py

```diff
@@ -442,19 +442,32 @@ def dropDb(self, dbName):
             _LOG.info('Database %s is in use by %d queries', dbName, len(qids))
             return False
 
-        nodes = self.wcss.getNodes()
+        czars, workers = self.wcss.getNodes()
+        nodes = [(node, 'czar') for node in czars] + [(node, 'worker') for node in workers]
         if not nodes:
             _LOG.error('Could not find any nodes to work on')
             return False
 
         # TODO we should do it in parallel for better scaling
         nDropped = 0
-        for node in nodes:
+        for node, nodeType in nodes:
 
             wmgr = node.wmgrClient()
             if dbName not in wmgr.databases():
                 _LOG.info('Database %r does not exist on node %r', dbName, node.name())
             else:
+                if nodeType == 'worker':
+                    # Try to remove this database from chunk inventory
+                    try:
+                        # Do not restart xrootd
+                        restart = False
+                        wmgr.xrootdUnregisterDb(dbName, restart)
+                        _LOG.info('Removed database %r from chunk registry on node %r', dbName, node.name())
+                    except ServerError as exc:
+                        if exc.code == 409:
+                            _LOG.info('Database %r is not in chunk registry on node %r', dbName, node.name())
+                        else:
+                            raise
                 try:
```

Comment:
The corresponding database in CSS will me marked with 'FAILED' status. What happens after that still needs to be defined, It's not clear how much auto-recovery we can do in this case. This more or less the same problem as what happens id DROP DATABASE fails in mysql server?